### PR TITLE
Select previous selected tables in the tablesList after choose of "Open Table in New Tab" context menu item

### PR DESCRIPTION
**Pull request by stychos**
What does this implement/fix? Explain your changes.
---------------------------------------------------
When users select "Open Table in New Tab" from context menu of the Tables List, they gets two tabs with the same table.

Does this close any currently open issues?
------------------------------------------
#191 

Any other comments?
-------------------
Not sure it's a right direction to resolve this, sometimes buggy.

Where has this been tested?
---------------------------
**Devices/Simulators:** Intel

**macOS Version:** 10.15.6

**Sequel-Ace Version:** 2.1.6rc1

### CODE REVIEW

**Comment by stychos** [comment]
By the way, now Sequel Ace / Sequel Pro often opens empty table too.

File: Source/SPTablesList.m [hunk 1]
```diff
@@ -147,10 +148,32 @@ - (void)awakeFromNib
...
+ * Not a menuDidOpen because it fires after tableViewSelectionDidChange which is responsible for DB View (and other things) refresh
+ */
+- (void)rightMouseDown:(NSEvent *)event
+{
+	isMenuOpened = true;
+}
+
+- (void)menuDidClose:(NSMenu *)menu {
+	isMenuOpened = false;
+}
```
Comment (Jason-Morcos) [inline Source/SPTablesList.m:167]: Looks like nothing calls this?

File: Source/SPTablesList.m [hunk 2]
```diff
@@ -1740,6 +1765,8 @@ - (BOOL)selectionShouldChangeInTableView:(nullable NSTableView *)aTableView
  */
 - (void)tableViewSelectionDidChange:(NSNotification *)aNotification
 {
+	if (!isMenuOpened) lastSelectedRows = [tablesListView selectedRowIndexes];
+
```
Comment (Jason-Morcos) [inline Source/SPTablesList.m:1769]: I'm a little confused, why do we need to track if the menu is open? Perhaps we should remember which table is selected distinctly from what row is selected? I don't really like the tracking-menu-is-open thing

**Comment by Jason-Morcos** [comment]
I noticed in finder, right clicking a list item puts an outline around it whereas left clicking an item fully highlights it. I think that's probably the right approach here - drawing an outline while something is being right-click-actioned and circumventing the default selection behavior during a right click.

File: Source/SPTablesList.m (hunk 1 above)
Comment (stychos) [reply Source/SPTablesList.m:167]: It's called automatically via NSMenuDelegate

File: Source/SPTablesList.m (hunk 2 above)
Comment (stychos) [reply Source/SPTablesList.m:1769]: Yes I think you right

**Comment by stychos** [comment]
As I understand here we have a deal with the table, and it autoselects an item under cursor on right click. Like with the icons view in the Finder. I am nearly ready to give up with this, can't find quick and easy solution. Sorry I started this PR too early, will need more time.

**Comment by Jason-Morcos** [comment]
@stychos I'm gonna go ahead and close this for now but definitely open to considering it later if ya figure it out! Would be nice to handle

**Comment by stychos** [comment]
Sure!

**Pull request closed without merging**